### PR TITLE
Fix auto attach state not returning after closing it

### PR DESCRIPTION
**Pull request by tintinthong**
https://linear.app/cardstack/issue/CS-6645/auto-attached-card-should-be-set-when-a-card-is-opened-even-if-that

### CODE REVIEW

File: packages/matrix/tests/messages.spec.ts [hunk 1]
```diff
@@ -690,9 +690,23 @@ test.describe('Room messages', () => {
...
+    await page
+      .locator(
+        `[data-test-stack-item-content] [data-test-cards-grid-item='${testCard1}']`,
+      )
+      .click();
+    await expect(page.locator(`[data-test-attached-card]`)).toHaveCount(1);
+    await page
+      .locator(`[data-test-stack-card='${testCard1}'] [data-test-close-button]`)
+      .click();
+    await expect(page.locator(`[data-test-attached-card]`)).toHaveCount(0);
```
Comment (tintinthong) [inline packages/matrix/tests/messages.spec.ts:708]: This is the only change in this file. The rest are formatting changes

File: packages/host/app/components/matrix/room.gts [hunk 2]
```diff
@@ -302,16 +292,13 @@ export default class Room extends Component<Signature> {
         return undefined;
       }
     }
-    this.setLastTopMostCard(topMostCard);
```
Comment (tintinthong) [inline packages/host/app/components/matrix/room.gts:305]: I dono why there was a set inside of a getter. 

EDIT: This motivated me to make a resource to react to the changes of the stack and maintain its own state of the auto attached card. More code but I think better?

File: packages/matrix/tests/messages.spec.ts (hunk 1 above)
Comment (backspace) [reply packages/matrix/tests/messages.spec.ts:708]: Thanks for drawing attention to this, I’m confused why this keeps happening 😞

**Review by backspace — APPROVED** [review]
Elegant approach with the resource! I confirmed this works:

screencast 2024-04-17 16-49-05

**Review by FadhlanR — APPROVED** [review]
Great!

**Pull request merged**